src/core/geometry/geom3/reTesselateCoplanarPolygons.js

```diff
@@ -7,332 +7,334 @@ const poly3 = require('../poly3')
+        vertices2d = []
+        numvertices = 0
+        minindex = -1
+      } else {
+        if (!(miny in topy2polygonindexes)) {
+          topy2polygonindexes[miny] = []
+        }
+        topy2polygonindexes[miny].push(polygonindex)
+      }
+    } // if(numvertices > 0)
+    // reverse the vertex order:
+    vertices2d.reverse()
+    minindex = numvertices - minindex - 1
+    polygonvertices2d.push(vertices2d)
+    polygontopvertexindexes.push(minindex)
+  }
+  const ycoordinates = []
+  for (let ycoordinate in ycoordinatetopolygonindexes) ycoordinates.push(ycoordinate)
+  ycoordinates.sort(fnNumberSort)
 
-    // Now we will iterate over all y coordinates, from lowest to highest y coordinate
-    // activepolygons: source polygons that are 'active', i.e. intersect with our y coordinate
-    //   Is sorted so the polygons are in left to right order
-    // Each element in activepolygons has these properties:
-    //        polygonindex: the index of the source polygon (i.e. an index into the sourcepolygons
-    //                      and polygonvertices2d arrays)
-    //        leftvertexindex: the index of the vertex at the left side of the polygon (lowest x)
-    //                         that is at or just above the current y coordinate
-    //        rightvertexindex: dito at right hand side of polygon
-    //        topleft, bottomleft: coordinates of the left side of the polygon crossing the current y coordinate
-    //        topright, bottomright: coordinates of the right hand side of the polygon crossing the current y coordinate
-    let activepolygons = []
-    let prevoutpolygonrow = []
-    for (let yindex = 0; yindex < ycoordinates.length; yindex++) {
-      let newoutpolygonrow = []
-      let ycoordinate_as_string = ycoordinates[yindex]
-      let ycoordinate = Number(ycoordinate_as_string)
+  // Now we will iterate over all y coordinates, from lowest to highest y coordinate
+  // activepolygons: source polygons that are 'active', i.e. intersect with our y coordinate
+  //   Is sorted so the polygons are in left to right order
+  // Each element in activepolygons has these properties:
+  //        polygonindex: the index of the source polygon (i.e. an index into the sourcepolygons
+  //                      and polygonvertices2d arrays)
+  //        leftvertexindex: the index of the vertex at the left side of the polygon (lowest x)
+  //                         that is at or just above the current y coordinate
+  //        rightvertexindex: dito at right hand side of polygon
+  //        topleft, bottomleft: coordinates of the left side of the polygon crossing the current y coordinate
+  //        topright, bottomright: coordinates of the right hand side of the polygon crossing the current y coordinate
+  let activepolygons = []
+  let prevoutpolygonrow = []
+  for (let yindex = 0; yindex < ycoordinates.length; yindex++) {
+    const newoutpolygonrow = []
```

Comment:
please switch to camel case for variables if possible 
ie `newoutpolygonrow` => `newOutPolygonRow`